pallets/pallet-nft/src/tests.rs

```diff
@@ -0,0 +1,210 @@
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NFTPending {
+			file_name,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_burned() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+		assert_ok!(Nft::burned(RuntimeOrigin::signed(ALICE), 1u32));
+
+		assert!(!Owners::<Runtime>::contains_key(1));
+		assert!(!Nfts::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftBurned {
+			owner: ALICE,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_transfer() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+		assert_eq!(Owners::<Runtime>::get(1), Some(ALICE));
+
+		assert_ok!(Nft::transfer(RuntimeOrigin::signed(ALICE), BOB, 1u32));
+
+		assert_eq!(Owners::<Runtime>::get(1), Some(BOB));
+		assert!(Nfts::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftTransferred {
+			from: ALICE,
+			to: BOB,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_approve_nft_pass() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+
+		assert_ok!(Nft::approve_nft(RuntimeOrigin::signed(FERDIE), 2u32, true));
+
+		assert_eq!(Owners::<Runtime>::get(2), Some(BOB));
+		assert!(Nfts::<Runtime>::contains_key(2));
+		assert!(!PendingNft::<Runtime>::contains_key(2));
+
+		System::assert_has_event(RuntimeEvent::Nft(Event::<Runtime>::NftMinted {
+			owner: BOB,
+			nft_id: 2u32,
+		}));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftAudited {
+			nft_id: 2u32,
+			approve: true,
+		}));
+	});
+}
+
+#[test]
+fn can_approve_nft_fail() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+		assert!(PendingNft::<Runtime>::contains_key(2));
+
+		assert_ok!(Nft::approve_nft(RuntimeOrigin::signed(FERDIE), 2u32, false));
+
+		assert!(!Owners::<Runtime>::contains_key(2));
+		assert!(!Nfts::<Runtime>::contains_key(2));
+		assert!(!PendingNft::<Runtime>::contains_key(2));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftAudited {
+			nft_id: 2u32,
+			approve: false,
+		}));
+	});
+}
+
+#[test]
+fn can_force_burn() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+
+		assert_ok!(Nft::force_burn(RuntimeOrigin::root(), 1u32));
+
+		assert!(!Owners::<Runtime>::contains_key(1));
+		assert!(!Nfts::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftBurned {
+			owner: ALICE,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn test_error_cases() {
+	default_test_ext().execute_with(|| {
+		let valid_file_name = vec![0x46, 0x49, 0x4C, 0x45];
+		let valid_data = vec![0x4E, 0x46, 0x54];
+		let invalid_file_name = vec![0x00; 260];
+		let invalid_data = vec![0x00; 1_001];
+
+		// Ferdie is auditor who can not create nft
+		assert_noop!(
+			Nft::request_mint(
+				RuntimeOrigin::signed(FERDIE),
+				valid_file_name.clone(),
+				valid_data.clone()
+			),
+			Error::<Runtime>::IncorrectRole
+		);
+		// File name is too large
```

Comment:
use "assert_eq" to show the max file and data size before asserting the error.
Also asser it is OK if the size is == the max (though this is tested else where, but this makes clear the limit is working as intended)